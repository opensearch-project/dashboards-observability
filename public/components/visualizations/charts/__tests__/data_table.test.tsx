/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { AgGridReact } from 'ag-grid-react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { TEST_VISUALIZATIONS_DATA } from '../../../../../test/event_analytics_constants';
import { DataTable } from '../data_table/data_table';

describe('Data table component', () => {
  configure({ adapter: new Adapter() });

  it('Renders data table component', async () => {
    const gridWrapper = shallow(<DataTable visualizations={TEST_VISUALIZATIONS_DATA} />);
    const agGridReactObj = gridWrapper.find(AgGridReact);
    agGridReactObj.simulate('gridReady');
    expect(agGridReactObj).toBeTruthy();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
