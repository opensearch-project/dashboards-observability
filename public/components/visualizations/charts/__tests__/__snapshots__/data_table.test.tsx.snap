// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data table component Renders data table component 1`] = `
<Fragment>
  <AgGridReact
    animateRows={true}
    columnDefs={
      Array [
        Object {
          "colId": "agent",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "agent",
          "headerName": "agent",
          "lockVisible": true,
        },
        Object {
          "colId": "bytes",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "bytes",
          "headerName": "bytes",
          "lockVisible": true,
        },
        Object {
          "colId": "clientip",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "clientip",
          "headerName": "clientip",
          "lockVisible": true,
        },
        Object {
          "colId": "event",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "event",
          "headerName": "event",
          "lockVisible": true,
        },
        Object {
          "colId": "extension",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "extension",
          "headerName": "extension",
          "lockVisible": true,
        },
        Object {
          "colId": "geo",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "geo",
          "headerName": "geo",
          "lockVisible": true,
        },
        Object {
          "colId": "host",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "host",
          "headerName": "host",
          "lockVisible": true,
        },
        Object {
          "colId": "index",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "index",
          "headerName": "index",
          "lockVisible": true,
        },
        Object {
          "colId": "ip",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "ip",
          "headerName": "ip",
          "lockVisible": true,
        },
        Object {
          "colId": "machine",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "machine",
          "headerName": "machine",
          "lockVisible": true,
        },
        Object {
          "colId": "memory",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "memory",
          "headerName": "memory",
          "lockVisible": true,
          "valueGetter": [Function],
        },
        Object {
          "colId": "message",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "message",
          "headerName": "message",
          "lockVisible": true,
        },
        Object {
          "colId": "phpmemory",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "phpmemory",
          "headerName": "phpmemory",
          "lockVisible": true,
        },
        Object {
          "colId": "referer",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "referer",
          "headerName": "referer",
          "lockVisible": true,
        },
        Object {
          "colId": "request",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "request",
          "headerName": "request",
          "lockVisible": true,
        },
        Object {
          "colId": "response",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "response",
          "headerName": "response",
          "lockVisible": true,
        },
        Object {
          "colId": "tags",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "tags",
          "headerName": "tags",
          "lockVisible": true,
        },
        Object {
          "colId": "timestamp",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "timestamp",
          "headerName": "timestamp",
          "lockVisible": true,
        },
        Object {
          "colId": "url",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "url",
          "headerName": "url",
          "lockVisible": true,
        },
        Object {
          "colId": "utc_time",
          "columnsMenuParams": Object {
            "suppressColumnExpandAll": true,
            "suppressColumnFilter": true,
            "suppressColumnSelectAll": true,
          },
          "field": "utc_time",
          "headerName": "utc_time",
          "lockVisible": true,
        },
      ]
    }
    defaultColDef={
      Object {
        "filter": undefined,
        "flex": 1,
        "headerHeight": 400,
        "minWidth": 100,
        "resizable": true,
        "sortable": true,
        "suppressMenu": false,
        "type": undefined,
      }
    }
    domLayout="autoHeight"
    onGridReady={[Function]}
    paginationPageSize={10}
    rowData={
      Array [
        Object {
          "double_per_ip_bytes": 11606,
          "host": "artifacts.opensearch.org",
          "ip_count": 176,
          "per_ip_bytes": 5803,
          "resp_code": "404",
          "sum_bytes": 1021420,
        },
        Object {
          "double_per_ip_bytes": 10100,
          "host": "www.opensearch.org",
          "ip_count": 111,
          "per_ip_bytes": 5050,
          "resp_code": "404",
          "sum_bytes": 560638,
        },
        Object {
          "double_per_ip_bytes": 0,
          "host": "artifacts.opensearch.org",
          "ip_count": 94,
          "per_ip_bytes": 0,
          "resp_code": "503",
          "sum_bytes": 0,
        },
        Object {
          "double_per_ip_bytes": 0,
          "host": "www.opensearch.org",
          "ip_count": 78,
          "per_ip_bytes": 0,
          "resp_code": "503",
          "sum_bytes": 0,
        },
        Object {
          "double_per_ip_bytes": 11526,
          "host": "cdn.opensearch-opensearch-opensearch.org",
          "ip_count": 43,
          "per_ip_bytes": 5763,
          "resp_code": "404",
          "sum_bytes": 247840,
        },
        Object {
          "double_per_ip_bytes": 0,
          "host": "cdn.opensearch-opensearch-opensearch.org",
          "ip_count": 34,
          "per_ip_bytes": 0,
          "resp_code": "503",
          "sum_bytes": 0,
        },
        Object {
          "double_per_ip_bytes": 8882,
          "host": "opensearch-opensearch-opensearch.org",
          "ip_count": 13,
          "per_ip_bytes": 4441,
          "resp_code": "404",
          "sum_bytes": 57735,
        },
        Object {
          "double_per_ip_bytes": 0,
          "host": "opensearch-opensearch-opensearch.org",
          "ip_count": 6,
          "per_ip_bytes": 0,
          "resp_code": "503",
          "sum_bytes": 0,
        },
      ]
    }
    rowHeight={35}
    suppressFieldDotNotation={true}
    suppressPaginationPanel={true}
  />
</Fragment>
`;
